packages/zowe-explorer/src/job/actions.ts

```diff
@@ -574,42 +574,41 @@ export async function filterJobs(jobsProvider: IZoweTree<IZoweJobTreeNode>): Pro
     return inputBox;
 }
 
-export async function filterSpools(
-    jobsProvider: IZoweTree<IZoweJobTreeNode>,
-    job: IZoweJobTreeNode,
-    zoweFileProvider: IZoweTree<IZoweNodeType>
-): Promise<vscode.InputBox> {
-    if (job["collapsibleState"] == 1) {
-        const spools = await getSpoolFiles(job);
-        const Spools = spools.map((spool) => {
-            const spoolNode = new Spool(
-                `${spool.stepname}:${spool.ddname} - ${spool["record-count"]}`,
-                vscode.TreeItemCollapsibleState.None,
-                job.getParent(),
-                job.getSession(),
-                spool,
-                job.job,
-                job.getParent()
+export async function filterSpools(jobsProvider: IZoweTree<IZoweJobTreeNode>, job: IZoweJobTreeNode): Promise<vscode.InputBox> {
+    try {
+        if (job["collapsibleState"] == vscode.TreeItemCollapsibleState.Collapsed) {
+            const Spools = (await getSpoolFiles(job)).map((spool) => {
+                const spoolNode = new Spool(
+                    `${spool.stepname}:${spool.ddname} - ${spool["record-count"]}`,
+                    vscode.TreeItemCollapsibleState.None,
+                    job.getParent(),
+                    job.getSession(),
+                    spool,
+                    job.job,
+                    job.getParent()
+                );
+                return spoolNode;
+            });
+            job.children = Spools;
+
+            await TreeViewUtils.expandNode(job, jobsProvider);
+            job.collapsibleState = vscode.TreeItemCollapsibleState.Expanded;
+            jobsProvider.refresh();
+        }
+
+        const actual_spools = job.children;
+        const inputBox = vscode.window.createInputBox();
+        inputBox.placeholder = localize("filterJobs.prompt.message", "Enter local filter...");
+        inputBox.onDidChangeValue((query) => {
+            query = query.toUpperCase();
+            job["children"] = actual_spools.filter((item) =>
+                `${item["spool"].stepname as string}:${item["spool"].ddname as string} - ${item["spool"]["record-count"] as string}`.includes(query)
             );
-            return spoolNode;
+            jobsProvider.refresh();
```

Comment:
We should refresh the `job` element here instead of the whole tree; otherwise, it will perform logic on other job nodes that are not a part of this filter operation.

This would also apply to line 571 in the `filterJobs` function - we can refresh the session (`level`) element for that case

Reply:
I tried both jobProvider.refresh and refreshElement(job) and don't see either of these working like I see the filter search at the session level.